index.html

```diff
@@ -65,10 +65,29 @@ <h2>preferCurrentTab</h2>
         </dt>
         <dd>
           <p>
-            When {{MediaDevices/getDisplayMedia()}} is called, if <code>preferCurrentTab</code> is
-            <code>true</code>, the user agent SHOULD present the current tab as the most prominent
-            option. Otherwise, no such requirement is made.
+            When {{MediaDevices/getDisplayMedia()}} is called, the user agent MUST run the following
+            steps:
           </p>
+          <ol>
+            <li>
+              <p>Let <var>constraints</var> be the method's first argument.</p>
+            </li>
+            <li>
+              If <code>constraints.preferCurrentTab</code> is <code>false</code>, abort these steps.
+            </li>
+            <li>
+              If <code>constraints.</code>{{DisplayMediaStreamConstraints/selfBrowserSurface}} is
+              {{SelfCapturePreferenceEnum/"exclude"}}, return a promise <a>rejected</a>
+              with a {{DOMException}} object whose {{DOMException/name}} attribute has the value
+              {{InvalidStateError}}.
```

Comment:
Current chromium implementation rejects with `TypeError`, not `DOMException`. Shall we update spec or implementation?

Reply:
On the one hand, surrounding code, which is associated with similar errors, rejects with TypeError.
On the other hand, DOMException of type InvalidStateError seems more appropriate to me.
What do you guys think is best?

Reply:
This part of the spec, which deals with some problematic constraint configurations, uses TypeError:
> If CS contains a member named advanced, return a promise [rejected](https://webidl.spec.whatwg.org/#reject) with a newly [created](https://webidl.spec.whatwg.org/#dfn-create-exception) [TypeError](https://webidl.spec.whatwg.org/#exceptiondef-typeerror).
>
> If CS contains a member whose name specifies a constrainable property applicable to [display surface](https://www.w3.org/TR/screen-capture/#dfn-display-surface)s, and whose value in turn is a dictionary containing a member named either min or exact, return a promise [rejected](https://webidl.spec.whatwg.org/#reject) with a newly [created](https://webidl.spec.whatwg.org/#dfn-create-exception) [TypeError](https://webidl.spec.whatwg.org/#exceptiondef-typeerror).
>
> If CS contains a member whose name specifies a constrainable property applicable to [display surface](https://www.w3.org/TR/screen-capture/#dfn-display-surface)s, and whose value in turn is a dictionary containing a member named max, and that member's value in turn is less than the constrainable property's [floor value](https://www.w3.org/TR/screen-capture/#dfn-floor-value), then let failedConstraint be the name of the member, let message be either undefined or an informative human-readable message, and return a promise [rejected](https://webidl.spec.whatwg.org/#reject) with a new OverconstrainedError created by calling OverconstrainedError(failedConstraint, message).

I am not sure why TypeError for that and DOMException for other errors, but possibly it's best to stay consistent and go with TypeError here too. Thoughts?

Reply:
(I've changed the PR to match and return TypeError.)